### PR TITLE
test: address HIGH test coverage gaps across all suites

### PR DESCRIPTION
**Pull request by alwynpan**
## Summary

- Add 50 new tests across unit, integration, frontend, and E2E to close HIGH-severity coverage gaps
- Total test count: 1347 → 1397 (+50 tests)
- All RBAC blind spots for assignment_manager role now covered
- Frontend client-side validation, schema, and component interaction gaps filled

### Backend unit (+5)
- `User.activate` model: success + DB error propagation
- `GET /auth/me`: deleted user (401) + DB error (500) paths
- `PUT /users/:id/group`: invalid non-UUID groupId (400)

### Integration (+17)
- `POST /users/send-setup-emails`: 7 tests (admin flow, auth, validation, edge cases)
- AM RBAC: can assign groups, cannot escalate roles, blocked from admin-only endpoints
- Disabled user: blocked from join and leave
- Locked system: leave blocked for regular users
- `maxMembers` boundary: exactly equal to current count succeeds
- Full group import-mappings: error recorded in response

### Frontend unit (+24)
- Login/Register: client-side schema validation (empty fields, invalid email)
- Login: loading/disabled button state during API call
- `App.jsx`: PublicRoute redirects authenticated users away from /login and /register
- `createUserSchema`: 10 tests (groupId null coercion, sendSetupEmail, required fields)
- `UserMenu`: AM role hides student ID field, dropdown toggle close

### E2E (+4)
- AM can unassign a user from a group
- Admin can change user role via edit modal
- Import users with "overwrite" conflict action
- Setup-type token first-login journey (set password → login)

## Test plan

- [x] Backend unit: 613 passed
- [x] Frontend unit: 533 passed
- [x] Integration: 158 passed
- [x] E2E: 93 passed
- [x] Lint: clean
- [x] Format: clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/tests/integration/users.test.js [hunk 1]
```diff
@@ -805,3 +805,193 @@ describe('POST /api/users/import', () => {
...
+      url: '/api/users',
+      headers: { authorization: `Bearer ${adminToken}` },
+      payload: {
+        username: 'pending2',
+        email: 'pending2@test.com',
+        firstName: 'P',
+        lastName: 'Two',
+        sendSetupEmail: false,
+      },
+    });
```
[the remark this answers is not in the document]
Comment (alwynpan) [reply backend/tests/integration/users.test.js]: Fixed in 60d4bb9 — added `expect(res1.statusCode).toBe(201)` and `expect(res2.statusCode).toBe(201)` assertions for both user-creation calls.

File: frontend/tests/unit/App.test.jsx [hunk 2]
```diff
@@ -101,4 +101,37 @@ describe('App', () => {
...
+
+  it('redirects authenticated user from /register to /dashboard when registration enabled', () => {
+    useAuth.mockReturnValue({
+      isAuthenticated: true,
+      loading: false,
+      isAdmin: false,
+      isAssignmentManager: false,
+      user: { username: 'member', role: 'normal_user' },
+      registrationEnabled: true,
+    });
```
[the remark this answers is not in the document]
Comment (alwynpan) [reply frontend/tests/unit/App.test.jsx:129]: Fixed in 60d4bb9 — changed `role: 'normal_user'` to `role: 'user'` in both redirect tests to match the actual role enum used throughout the app.
